Jenkinsfile

```diff
@@ -37,325 +38,552 @@ pipeline {
+              allOf {
+                expression { return env.CHANGE_ID != null}
+                expression { return env.GIT_PREVIOUS_COMMIT != null }
+                expression { return env.CHANGE_TARGET ==~ /(master|develop|trunk)/ }
+                expression { return params.Merge_PR }
               }
             }
           }
+          agent { label 'armv8' }
+          steps {
+            script {
+              def debugBuild = load ".jenkinsci/debug-build.groovy"
+              def testSelect = load ".jenkinsci/test-launcher.groovy"
+              debugBuild.doTestStep(testSelect.chooseTestType())
+            }
+          }
         }
-        stage('ARMv8') {
+        stage('MacOS') {
           when {
             beforeAgent true
-            expression { return params.ARMv8 }
+            anyOf {
+              allOf {
+                expression { return env.CHANGE_ID }
+                not { expression { return env.GIT_PREVIOUS_COMMIT } }
+              }
+              allOf {
+                expression { return env.CHANGE_ID != null}
+                expression { return env.GIT_PREVIOUS_COMMIT != null }
+                expression { return env.CHANGE_TARGET ==~ /(master|develop|trunk)/ }
+                expression { return params.Merge_PR }
+              }
+              expression { return params.MacOS }
+            }
           }
-          agent { label 'armv8' }
+          agent { label 'mac' }
           steps {
             script {
-              def releaseBuild = load ".jenkinsci/release-build.groovy"
-              releaseBuild.doReleaseBuild()
+              def macDebugBuild = load ".jenkinsci/mac-debug-build.groovy"
+              def testSelect = load ".jenkinsci/test-launcher.groovy"
+              macDebugBuild.doTestStep(testSelect.chooseTestType())
             }
           }
-          post {
-            always {
-              script {
-                post = load ".jenkinsci/linux-post-step.groovy"
-                post.linuxPostStep()
+        }
+      }
+    }
+    stage('Post-Coverage') {
+      when {
+        anyOf {
+          expression { params.Coverage }  // by request
+          allOf {
+            expression { return env.CHANGE_ID }
+            not { expression { return env.GIT_PREVIOUS_COMMIT } }
+          }
+          allOf {
+            expression { return env.CHANGE_ID != null}
+            expression { return env.GIT_PREVIOUS_COMMIT != null }
+            expression { return env.CHANGE_TARGET ==~ /(master|develop|trunk)/ }
+            expression { return params.Merge_PR }
+          }
+          allOf {
+            expression { return params.BUILD_TYPE == 'Debug' }
+            expression { return env.GIT_LOCAL_BRANCH ==~ /master/ }
+          }
+        }
+      }
+      parallel {
+        stage('lcov_cobertura') {
+          steps {
+            script {
+              def cov_platform = ''
+              if (params.Linux) {
```

Comment:
Maybe implement this platform selection as a separate function?

Reply:
implemented in the `.jenkinsci/choose-platform.groovy`